src/vm.rs

```diff
@@ -1438,6 +1464,31 @@ impl Machine {
                         unreachable!()
                     }
                 }
+                Opcode::IsTaggedTuple => {
+                    debug_assert_eq!(ins.args.len(), 4);
+                    let fail = self.expand_arg(context, &ins.args[0]).to_u32();
+                    let reg = self.expand_arg(context, &ins.args[1]);
+                    let n = self.expand_arg(context, &ins.args[2]).to_u32();
+                    let atom = self.expand_arg(context, &ins.args[3]);
+                    if let Value::Tuple(t) = reg {
+                        let arity = unsafe { (**t).len() };
+                        if arity == 0 || arity != (n as usize) {
+                            op_jump!(context, fail);
+                        } else {
+                            let elem = unsafe {
```

Comment:
I think

```rust
let elem = unsafe { &(**t)[0] }
```

should work, that way we avoid doing a clone.

Reply:
I've pushed another commit that changes this.

I'd originally copied this from the [implementation of `GetTupleElement`](https://github.com/archSeer/enigma/blob/b84eed59b26ec618cd87df804e62bc45607d2aa0/src/vm.rs#L964-L967) because I don't know Rust well enough yet to know when to do one thing versus the other in a lot of cases.